genie/genie-controller.go

```diff
@@ -324,13 +325,14 @@ func ParsePodAnnotationsForMultiIPPrefs(client *kubernetes.Clientset, k8sArgs ut
 }
 
 //  parseCNIAnnotations parses pod yaml defintion for "cni" annotations.
-func parseCNIAnnotations(annot map[string]string, client *kubernetes.Clientset, k8sArgs utils.K8sArgs) ([]string, error) {
+func parseCNIAnnotations(annot map[string]string, client *kubernetes.Clientset, k8sArgs utils.K8sArgs, conf utils.NetConf) ([]string, error) {
 	var finalAnnots []string
 
-	if len(annot) == 0 {
-		fmt.Fprintf(os.Stderr, "CNI Genie no annotations is given! Default plugin is weave! annot is %V\n", annot)
-		finalAnnots = []string{"weave"}
-	} else if strings.TrimSpace(annot["cni"]) == "" {
+	if len(annot) == 0 || strings.TrimSpace(annot["cni"]) == "" {
+		if conf.Default != "" {
```

Comment:
I'm on the east coast, happy to chat just about any time. The Slack you mentioned is a Huawei specific slack, doesn't have a public invite. I'm happy to chat on https://kubernetes.slack.com or Hangout.